File: platform/sdk/nodejs/examples/t3-01-create-upload.js
const { CloudClient } = require("../xuanji-sdk-cloud");

function main() {
  const client = new CloudClient();
  const r = client.createMultipartUpload("my-bucket", "data/large-file.bin");
  if (!r.ok) throw new Error("create failed: " + r.message);
  const uid = r.upload_id;
  if (!uid || uid.length === 0) throw new Error("upload_id empty");
  if (!uid.startsWith("mpu-")) throw new Error("upload_id must start with mpu-: " + uid);
  console.log("XJ-OK: t3-01-create-upload id=" + uid);
}

if (require.main === module) main();
module.exports = main;

File: platform/sdk/python/examples/t3_01_create_upload.py
import sys, os
sys.path.insert(0, os.path.join(os.path.dirname(__file__), ".."))
from xuanji_sdk_cloud import CloudClient

def main():
    client = CloudClient()
    r = client.createMultipartUpload("my-bucket", "data/large-file.bin")
    if not r["ok"]:
        raise RuntimeError("create failed: " + str(r))
    uid = r["upload_id"]
    if not uid or len(uid) == 0:
        raise RuntimeError("upload_id empty")
    if not uid.startswith("mpu-"):
        raise RuntimeError("upload_id must start with mpu-: " + uid)
    print("XJ-OK: t3_01_create_upload id=" + uid)

if __name__ == "__main__":
    main()
